src/github.com/matrix-org/dendrite/clientapi/writers/createroom.go

```diff
@@ -117,12 +132,136 @@ func createRoom(req *http.Request, roomID string) util.JSONResponse {
 	// This differs from Synapse slightly. Synapse would vary the ordering of 3-7
 	// depending on if those events were in "initial_state" or not. This made it
 	// harder to reason about, hence sticking to a strict static ordering.
+	// TODO: Synapse has txn/token ID on each event. Do we need to do this here?
+	emptyString := ""
+	eventsToMake := []fledglingEvent{
+		{"m.room.create", &emptyString, events.CreateContent{Creator: userID}},
+		{"m.room.member", &userID, events.MemberContent{Membership: "join"}}, // TODO: Set avatar_url / displayname
+		{"m.room.power_levels", &emptyString, events.InitialPowerLevelsContent(userID)},
+		// TODO: m.room.canonical_alias
+		{"m.room.join_rules", &emptyString, events.JoinRulesContent{"public"}},                 // FIXME: Allow this to be changed
+		{"m.room.history_visibility", &emptyString, events.HistoryVisibilityContent{"joined"}}, // FIXME: Allow this to be changed
+		// TODO: m.room.guest_access
+		// TODO: Other initial state items
+		// TODO: m.room.name
+		// TODO: m.room.topic
+		// TODO: invite events
+		// TODO: 3pid invite events
+		// TODO m.room.aliases
+	}
+
+	authEvents := authEventProvider{builtEventMap}
+	for i, e := range eventsToMake {
+		depth := i + 1 // depth starts at 1
+
+		builder := gomatrixserverlib.EventBuilder{
+			Sender:   userID,
+			RoomID:   roomID,
+			Type:     e.Type,
+			StateKey: e.StateKey,
+			Depth:    int64(depth),
+		}
+		builder.SetContent(e.Content)
+		if i > 0 {
+			builder.PrevEvents = []gomatrixserverlib.EventReference{builtEvents[i-1].EventReference()}
+		}
+		ev, err := buildEvent(&builder, builtEventMap, cfg)
+		if err != nil {
+			return util.ErrorResponse(err)
+		}
+
+		if err := gomatrixserverlib.Allowed(*ev, &authEvents); err != nil {
+			return util.ErrorResponse(err)
+		}
+
+		builtEventMap[common.StateKeyTuple{e.Type, *e.StateKey}] = ev
```

Comment:
It can't be nil. fledglingEvent allows it to be nil because.... okay fine, it shouldn't be nullable.

Done. https://github.com/matrix-org/dendrite/pull/32/commits/b6bc570095e22ee23860550d0697b11de308c857